AK/Math.h

```diff
@@ -532,6 +532,97 @@ using Hyperbolic::cosh;
 using Hyperbolic::sinh;
 using Hyperbolic::tanh;
 
+template<Integral I, FloatingPoint P>
+ALWAYS_INLINE I round_to(P value)
+{
+#if ARCH(I386) || ARCH(X86_64)
+    Conditional<sizeof(I) >= sizeof(i16), I, i16> ret;
+    if constexpr (sizeof(I) == sizeof(i64)) {
+        asm(
+            "fistpll %0"
+            : "=m"(ret)
+            : "t"(value)
+            : "st");
+    }
+    if constexpr (sizeof(I) == sizeof(i32)) {
+        asm(
+            "fistpl %0"
+            : "=m"(ret)
+            : "t"(value)
+            : "st");
+    } else {
+        asm(
+            "fistps %0"
+            : "=m"(ret)
+            : "t"(value)
+            : "st");
+    }
+    return ret;
+#else
+    if constexpr (IsSame<P, long double>)
+        return (I)llrintl(value);
+    if constexpr (IsSame<P, double>)
+        return (I)llrint(value);
+    if constexpr (IsSame<P, float>)
+        return (I)llrintf(value);
+#endif
+}
+
+#ifdef __SSE__
+template<Integral I>
+ALWAYS_INLINE I round_to(float value)
+{
+#    if ARCH(I386) || ARCH(X86_64)
+#        if ARCH(X86_64)
+    if constexpr (sizeof(I) == sizeof(i64)) {
```

Comment:
Maybe `#elif` instead of duplicating the condition here? Found this a bit confusing to read.

Reply:
Well This is not an explicit else,
This part is x64 only, while the following is x86 and x64,
I will reorder the statements, to make it more readeable though

Reply:
It should now be a bit clearer why there is this distinction

Reply:
Yeah, this is easier to follow. Thanks!